src/fstab-generator/fstab-generator.c

```diff
@@ -451,14 +451,39 @@ static int add_mount(
                         return r;
         }
 
-        if (!(flags & NOAUTO) && !(flags & AUTOMOUNT)) {
-                r = generator_add_symlink(dest, post,
-                                          (flags & NOFAIL) ? "wants" : "requires", name);
-                if (r < 0)
-                        return r;
-        }
+        if (!(flags & AUTOMOUNT)) {
+                if (!(flags & NOAUTO)) {
+                        r = generator_add_symlink(dest, post,
+                                                  (flags & NOFAIL) ? "wants" : "requires", name);
+                        if (r < 0)
+                                return r;
+                } else {
+                        _cleanup_strv_free_ char **units = NULL;
+                        char **s;
 
-        if (flags & AUTOMOUNT) {
+                        r = fstab_extract_values(opts, "x-systemd.wanted-by", &units);
+                        if (r < 0)
+                                return r;
+
+                        STRV_FOREACH(s, units) {
+                                r = generator_add_symlink(dest, *s, "wants", name);
+
+                                if (r < 0)
```

Comment:
spurious newline. We generally try to keep lines and their error handling together, hence the newline shouldn't be there